nx-dev/feature-ai/src/lib/feature-ai.tsx

```diff
@@ -0,0 +1,103 @@
+import { ReactNode, useState } from 'react';
+import { Button } from '@nx/nx-dev/ui-common';
+import { sendCustomEvent } from '@nx/nx-dev/feature-analytics';
+
+import { renderMarkdown } from '@nx/nx-dev/ui-markdoc';
+import { nxDevDataAccessAi } from '@nx/nx-dev/data-access-ai';
+
+export function FeatureAi() {
+  const [finalResult, setFinalResult] = useState('' as ReactNode);
+  const [error, setError] = useState(null);
+  const [query, setSearchTerm] = useState('');
+  const [loading, setLoading] = useState(false);
+
+  const warning = `
+  {% callout type="warning" title="Always double check!" %}
+  This feature is still in Alpha.
+  The results may not be accurate, so please always double check with our documentation.
+{% /callout %}
+
+  `;
+
+  const handleSubmit = async () => {
+    setLoading(true);
+    let completeText = '';
+    sendCustomEvent('ai_query', 'ai', query);
+    try {
+      completeText = await nxDevDataAccessAi(query);
+      setLoading(false);
+    } catch (error) {
+      console.error(error);
+      setError(error as any);
+      setLoading(false);
+    }
+    setFinalResult(
+      renderMarkdown(warning + completeText, { filePath: '' }).node
+    );
+  };
+  return (
+    <div className="p-2 mx-auto flex h-full w-full flex-col">
+      <div className="w-full flex">
+        <input
+          id="search"
+          name="search"
+          disabled={loading}
+          className="block w-full rounded-md border border-slate-300 bg-white py-2 pl-10 pr-3 text-sm placeholder-slate-500 transition focus:placeholder-slate-400 dark:border-slate-900 dark:bg-slate-700"
+          placeholder="What do you want to know?"
+          onChange={(event) => setSearchTerm(event.target.value)}
+          onKeyDown={(event) => {
+            if (event.keyCode === 13 || event.key === 'Enter') {
+              handleSubmit();
+            }
+          }}
+          type="search"
+        />
+        <Button
+          variant="primary"
+          size="small"
+          disabled={loading}
+          onClick={() => handleSubmit()}
+        >
+          Ask
+        </Button>
+      </div>
+      {loading ? (
+        <div className="p-4 max-w-none">
+          <h1>Thinking...</h1>
+        </div>
+      ) : null}
+      {finalResult && !error ? (
+        <>
+          <div className="p-4 max-w-none prose prose-slate dark:prose-invert">
+            {finalResult}
+          </div>
+          <div>
+            <Button
+              variant="primary"
+              size="small"
+              onClick={() => sendCustomEvent('ai_feedback', 'ai', 'good')}
+            >
+              Answer was helpful{' '}
+              <span role="img" aria-label="thumbs-up">
+                👍
+              </span>
+            </Button>
+            <Button
+              variant="primary"
+              size="small"
+              onClick={() => sendCustomEvent('ai_feedback', 'ai', 'bad')}
```

Comment:
does this look ok? @bcabanes

Reply:
Can we send the query and result along with the feedback?